packages/pwa-kit-extension-sdk/bin/list-overrides.js

```diff
@@ -0,0 +1,195 @@
+    }
+}
+
+/**
+ * Groups files by their source extension
+ * @param {string[]} files - List of files to group
+ * @param {OverrideStatsEntry[]} rawStats - Raw stats data
+ * @returns {FilesByExtension} Files grouped by extension
+ */
+const groupFilesByExtension = (files, rawStats) => {
+    const filesByExtension = {}
+
+    // Create a map of original paths to their stats entries
+    const originalToStats = {}
+    rawStats.forEach((entry) => {
+        originalToStats[entry.original] = entry
+    })
+
+    files.forEach((file) => {
+        // Get the stats entry for this file
+        const statsEntry = originalToStats[file]
+
+        if (!statsEntry) {
+            console.warn(chalk.yellow(`Warning: No stats entry found for ${file}`))
+            return
+        }
+
+        // Use the sourceExtension property from the stats entry
+        const extensionName = statsEntry.sourceExtension || 'other'
+
+        // Extract the relative path (everything after src/)
+        const srcIndex = file.indexOf(`${path.sep}src${path.sep}`)
+        const relativePath =
+            srcIndex !== -1
+                ? file.substring(srcIndex + 5).replace(/\\/g, '/') // Normalize path separators to forward slashes
+                : file.replace(/\\/g, '/')
+
+        // Initialize the array for this extension if it doesn't exist
+        if (!filesByExtension[extensionName]) {
+            filesByExtension[extensionName] = []
+        }
+
+        // Add the file to the appropriate extension group
+        filesByExtension[extensionName].push(relativePath)
+    })
+
+    return filesByExtension
+}
+
+/**
+ * Main function to list overridable files
+ */
+const listOverrides = () => {
+    try {
+        console.log(chalk.cyan('Listing overridable files...\n'))
+
+        const statsData = readOverrideStats()
+
+        if (!statsData || !statsData.overridableFiles || !statsData.rawStats) {
+            console.error(chalk.red('No override stats found. Run a build first.'))
+            process.exit(1)
+        }
+
+        const filesByExtension = groupFilesByExtension(
+            statsData.overridableFiles,
+            statsData.rawStats
+        )
+
+        // Print files grouped by extension
+        console.log(chalk.cyan('\nOverridable files by extension:'))
+        Object.keys(filesByExtension)
+            .sort()
+            .forEach((extension) => {
+                console.log(`\n${chalk.bold(extension)}:`)
+                filesByExtension[extension].sort().forEach((file) => {
+                    console.log(`  ${file}`)
+                })
+            })
+
+        // Check for unused overrides
+        checkUnusedOverrides(statsData)
+    } catch (error) {
+        console.error(chalk.red(`Error: ${error.message}`))
+        process.exit(1)
+    }
+}
+
+// Export for use in pwa-kit-dev
+module.exports = {
+    listOverrides
+}
+
+// Allow direct execution
+if (require.main === module) {
+    listOverrides()
+}
```

Comment:
You probably won't require this module.exports unless you are importing it else where, but I don't think you should have to. You might think about changing the name of the listoverrides function to main as that is typically the convension for scripts. If you need to pass args to it to change it's behaviour, that is where you would have your conditions on what methods to call.